pkg/apps/gitops.go

```diff
@@ -166,9 +167,38 @@ func (o *GitOpsOptions) DeleteApp(app string, alias string, autoMerge bool) erro
 
 // GetApps retrieves all the apps information for the given appNames from the repository and / or the CRD API
 func (o *GitOpsOptions) GetApps(appNames map[string]bool, expandFn func([]string) (*v1.AppList, error)) (*v1.AppList, error) {
-	dir, _, _, _, err := gits.ForkAndPullRepo(o.DevEnv.Spec.Source.URL, o.EnvironmentsDir, o.DevEnv.Spec.Source.Ref, "master", o.GitProvider, o.Gitter, "")
+	dir := o.EnvironmentCloneDir
+	if dir == "" {
+		tmpDir, err := ioutil.TempDir("", "get-apps-")
+		if err != nil {
+			return nil, err
+		}
+		defer os.RemoveAll(tmpDir)
+		dir = tmpDir
+	}
+
+	gitInfo, err := gits.ParseGitURL(o.DevEnv.Spec.Source.URL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "parsing dev env repo URL %s", o.DevEnv.Spec.Source.URL)
+	}
+
+	providerInfo, err := o.GitProvider.GetRepository(gitInfo.Organisation, gitInfo.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "determining git provider information for %s", o.DevEnv.Spec.Source.URL)
+	}
+	cloneUrl := providerInfo.CloneURL
+	userDetails := o.GitProvider.UserAuth()
+	originFetchURL, err := o.Gitter.CreateAuthenticatedURL(cloneUrl, &userDetails)
```

Comment:
:-( - this is bad. We should not use this approach. Sad to see that this proliferates now outside ForkAndPullRepo

Reply:
I don't disagree, but I couldn't figure out how to ensure that `ForkAndPullRepo` _doesn't_ fork (it seems to fork if the current user is different from the owner of the repo) and I needed an authenticated clone URL, so...

Reply:
Sure

Reply:
Though I'd be fine with adding an argument to `ForkAndPullRepo` to explicitly block forking and just using it here.